### ISSUE TITLE
Update to CodeMirror 6

### ISSUE THREAD

**Issue by hildjj** [peggyjs/peggy#296]
The [online page](https://peggyjs.org/online.html) currently uses an old version of [CodeMirror 5](https://codemirror.net/5/).  We would like to [upgrade](https://codemirror.net/docs/migration/) to CodeMirror 6.

This would be a great first bug if you've worked with CM6 before, and would like to contribute to Peggy.  It's likely that the `pegjs` language mode for CM5 will also need to be rewritten for this to work.  In that case, we'll be happy to host a new project in the [PeggyJS](https://github.com/peggyjs) GitHub org, and give you control over that repo.

**Comment by polkovnikov-ph** [comment]
1. Monaco editor isn't really that hard to set up. Why not switch to an editor that can support something like this? (ugliest [code](https://github.com/polkovnikov-ph/grammar-debug) alert; press `Next` and hover in either column);
![image](https://user-images.githubusercontent.com/2983969/173264657-4a6636a0-1a5c-4635-b2f5-db6bde85c97e.png)
2. Would you consider a PR with more serious rewrite for the `/online` page? I probably know [where](https://github.com/polkovnikov-ph/pegjs-website) the idea of CodeMirror came from, but probably could do better this time.
![image](https://user-images.githubusercontent.com/2983969/173265273-c4d65525-3bef-40d6-95ba-c6107aaaf6e5.png)

**Comment by hildjj** [comment]
Yes, I'd look at a PR that switched to something that isn't CodeMirror.  I'm not tied to CM at all, and maintenance costs matter to me, so simpler is better particularly for non-core functionality.

**Comment by Mingun** [comment]
Yeah, I was actually also thinking, would not switch to monaco?

**Comment by hildjj** [comment]
I am strongly in favor of moving to Monaco, after having tried it on another project.
